src/containers/AssetForm.js

```diff
@@ -422,6 +405,32 @@ class AssetForm extends Component {
   }
 }
 
-const mapDispatchToProps = { handleMessage: snackbarOpen };
+AssetForm.propTypes = {
+  match: PropTypes.object.isRequired,
+  history: PropTypes.object.isRequired,
+  snackbarOpen: PropTypes.func.isRequired,
+  getAsset: PropTypes.func.isRequired,
+  createAsset: PropTypes.func.isRequired,
+  updateAsset: PropTypes.func.isRequired,
+  assetUrl: PropTypes.string,
+  asset: PropTypes.object,
+};
+
+const mapDispatchToProps = { snackbarOpen, getAsset, createAsset, updateAsset };
+
+const mapStateToProps = ({ assets } , props) => {
+
+  let assetUrl = assets.url; // for resolving the url post-save
+  if (props.match.url !== '/asset/create') {
```

Comment:
Since the only prop we use is match could we just make the signature
```js
const mapStateToProps = ({ assets } , { match }) => {
```
and have this line be
```js
if (match.url !== '/asset/create') {
```